src/libstd/panicking.rs

```diff
@@ -149,6 +149,18 @@ impl<'a> PanicInfo<'a> {
         self.payload
     }
 
+    /// Returns the payload as a slice if it is a `&'static str` or `String`.
```

Comment:
Maybe a more future proof approach would be to have something like `Path::display`, i.e. a method returning an opaque type implementing `Display`?